packages/cojson-storage/src/syncUtils.ts

```diff
@@ -47,60 +40,16 @@ export function getDependedOnCoValues(
   header: CojsonInternalTypes.CoValueHeader,
   contentMessage: CojsonInternalTypes.NewContentMessage,
 ) {
-  const deps = new Set<RawCoID>();
-
-  for (const sessionID of Object.keys(contentMessage.new) as SessionID[]) {
-    const accountId = cojsonInternals.accountOrAgentIDfromSessionID(sessionID);
-
-    if (
-      cojsonInternals.isAccountID(accountId) &&
-      accountId !== contentMessage.id
-    ) {
-      deps.add(accountId);
-    }
-  }
-
-  if (header.ruleset.type === "group") {
-    /**
-     * Collect all the signing keys inside the transactions to list all the
-     * dependencies required to correctly access the CoValue.
-     */
-    for (const sessionEntry of Object.values(contentMessage.new)) {
-      for (const tx of sessionEntry.newTransactions) {
-        if (tx.privacy !== "trusting") continue;
-
-        const changes = safeParseChanges(tx.changes);
-        for (const change of changes) {
-          if (
-            change &&
-            typeof change === "object" &&
-            "op" in change &&
-            change.op === "set" &&
-            "key" in change &&
-            change.key
-          ) {
-            const key = cojsonInternals.getGroupDependentKey(change.key);
-
-            if (key) {
-              deps.add(key);
-            }
-          }
-        }
-      }
-    }
-  }
-
-  if (header.ruleset.type === "ownedByGroup") {
-    deps.add(header.ruleset.group);
-  }
-
-  return deps;
-}
-
-function safeParseChanges(changes: Stringified<JsonValue[]>) {
-  try {
-    return cojsonInternals.parseJSON(changes);
-  } catch (e) {
-    return [];
-  }
+  const id = contentMessage.id;
+  const sessionIDs = Object.keys(contentMessage.new) as SessionID[];
+  const transactions = Object.values(contentMessage.new).map(
+    (entry) => entry.newTransactions,
+  );
+
+  return cojsonInternals.getDependedOnCoValuesFromRawData(
```

Comment:
Since this util is now compatible with content messages we can reuse the same logic from cojson to process deps